examples/model-score/app.py

```diff
@@ -0,0 +1,305 @@
+        else:
+            # This approach works well if you know that input.refresh() is likely to be
+            # a longer interval than the underlying changing data source (df()). If not,
+            # then this can cause downstream reactives to be invalidated when they
+            # didn't need to be.
+            reactive.invalidate_later(refresh)
+            with reactive.isolate():
+                return df()
+
+    @reactive.Calc
+    def timeframe_df():
+        """
+        Returns rows from the database within the specified time range. Notice that we
+        implement the business logic as a separate function (read_time_period), so it's
+        easier to reason about and test.
+        """
+        start, end = input.timerange()
+        return read_time_period(start, end)
+
+    @reactive.Calc
+    def filtered_df():
+        """
+        Return the data frame that should be displayed in the app, based on the user's
+        input. This will be either the latest rows, or a specific time range. Also
+        filter out rows for models that the user has deselected.
+        """
+        data = recent_df() if input.timeframe() == "Latest" else timeframe_df()
+
+        # Filter the rows so we only include the desired models
+        return data[data["model"].isin(input.models())]
+
+    @reactive.Calc
+    def filtered_model_names():
+        return filtered_df()["model"].unique()
+
+    @output
+    @render.ui
+    def value_boxes():
+        data = filtered_df()
+        models = data["model"].unique().tolist()
+        scores_by_model = {
+            x: data[data["model"] == x].iloc[-1]["score"] for x in models
+        }
+        # Round scores to 2 decimal places
+        scores_by_model = {x: round(y, 2) for x, y in scores_by_model.items()}
+
+        return x.ui.layout_column_wrap(
+            "135px",
+            *[
+                # For each model, return a value_box with the score, colored based on
+                # how high the score is.
+                x.ui.value_box(
+                    model,
+                    ui.h2(score),
+                    theme_color="success"
+                    if score > THRESHOLD_MID
+                    else "warning"
+                    if score > THRESHOLD_LOW
+                    else "danger",
+                )
+                for model, score in scores_by_model.items()
+            ],
+            fixed_width=True,
+        )
+
+    @output
+    @render_plotly_streaming(recreate_key=filtered_model_names, update="data")
+    def plot_timeseries():
+        """
+        Returns a Plotly Figure visualization. Streams new data to the Plotly widget in
+        the browser whenever filtered_df() updates, and completely recreates the figure
+        when filtered_model_names() changes (see recreate_key=... above).
+        """
+        fig = px.line(
+            filtered_df(),
+            x="time",
+            y="score",
+            labels=dict(score="auc"),
```

Comment:
After thinking about it I'm not sure we should use AUC for this. While AUC can vary from 0-1, it's pretty weird to have a number lower than 0.5 because that indicates that the model is worse than guessing. Maybe we should change these to "accuracy".